### PR TITLE
Check graphs for instances

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?188655

Graph config might contains complex calculation and needs to be checked for instance references.

@dannyroberts / @proteusvacuum

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/xml_models.py [hunk 1]
```diff
@@ -603,12 +603,20 @@ def get_all_xpaths(self):
...
+            result = set()
+            for config in configuration.configs:
+                result.add(config.xpath_function)
+            return result
+
         for field in self.fields:
             if field.template.form == 'graph':
                 s = etree.tostring(field.template.node)
                 template = load_xmlobject_from_string(s, xmlclass=GraphTemplate)
+                result = result.union(_get_graph_config_xpaths(template.graph.configuration))
```
Comment (dannyroberts) [inline corehq/apps/app_manager/suite_xml/xml_models.py]: I like `result.update(...)` better; more analogous to dict.update/list.extend. (It's also trivially more efficient by not creating a new set, though that doesn't really matter in this case.)

**Comment by dannyroberts** [comment]
Nice, will wait for tests

File: corehq/apps/app_manager/suite_xml/xml_models.py (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/xml_models.py]: makes sense, fixed

Comment (dannyroberts) [inline corehq/apps/app_manager/suite_xml/xml_models.py]: :+1:

**Comment by czue** [comment]
nice! this totally tripped me up on the burpee app

**Comment by orangejenny** [comment]
tests pass

**Pull request merged**